### PR TITLE
Milestone5

### PR DESCRIPTION
**Pull request by Arinpe**
Here, we created the first working of friendship, users can create a post and send friendship requests to other users

### CODE REVIEW

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
# Status: Changes required👈

Hey coders, you have done a good job so far. I'm really impressed especially with the model associations. Let's just fix few issues before we can move.

- [x] User can see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Currently, I can't see a friendship request sent to me, instead I see a button to invite for friendship to the same user who sent a request to me (another user). Have a look at what is happening in the database from the image below.

friendship

- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). unfortunately, a user can see posts of other users who are not his/her friends.
- [x] You should add tests for other models and controllers as well.

<hr>

**[OPTIONAL]**

- [ ] You can add a button to delete/unfriend an existing friend.

In case you have a question about the review, feel free to send a direct message on slack by searching me as Elisha kyakopo.

Cheers🎉💪✔
Happy coding!

Elisha

File: app/models/friendship.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Friendship < ApplicationRecord
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+  validates :user, presence: true
+  validates :friend, presence: true
```
Comment (carlos-ssh) [inline app/models/friendship.rb:5]: - User model needs: ```has_many :friendships```

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hello Team 👋🏽
This project is really nice! 💪🏽🥷🏽👨🏽‍💻

Kindly help me with this changes:
- Post is always displayed with content, author, comments and likes. (I can't see Posts from other users).
- Would be great to have Negative tests examples using ```to not_eq```

- Please consider the code commented below: 

_**Happy Coding!**_  🖥🖱
@carlos-ssh

File: app/models/friendship.rb (hunk 1 above)
Comment (safafa) [reply app/models/friendship.rb:5]: I think we implemented this the User model actually has_many friendships. I cannot see where the problem is. Can you please give more explanation?

**Comment by safafa** [comment]
@carlos-ssh Can you please give more explanation considering this request : "Post is always displayed with content, author, comments and likes. (I can't see Posts from other users)."  Because in my side I can see posts written by logged-in user and their friends.

**Comment by carlos-ssh** [comment]
> @carlos-ssh Can you please give more explanation considering this request : "Post is always displayed with content, author, comments and likes. (I can't see Posts from other users)." Because in my side I can see posts written by logged-in user and their friends.

Hi @safafa 👋🏽 When I try to see posts from other users, just see my posts. 
friends_timeline

**Comment by safafa** [comment]
@carlos-ssh  Here I'm logged in as Safa and loubna and detola are my friends I can see their posts on my timeline
Screenshot 2021-06-17 at 13-31-13 RorSocialScaffold

**Comment by carlos-ssh** [comment]
Ok it's nice!, if your project is working well locally.
_one recommendation: try to remove it from your computer and install it again from GitHub and then try to do a last check, and then send a code review again!_ 
Greetings.

_Happy Coding!_

**Review by carlos-ssh — APPROVED** [review]
**CONGRATULATIONS ‼️**
STATUS: Approved 🍾🍾🍾

Hello Team 👋🏽👋🏽👋🏽
You can merge and continue working in your projects! 

_Happy Coding!_
@carlos-ssh

**Pull request merged**